torch/csrc/jit/codegen/cuda/fusion_segmenter.h

```diff
@@ -52,7 +52,7 @@ class TORCH_CUDA_CU_API SegmentedGroup {
 
   //! Checks if this group is used any where in the segmented fusion
   bool isConnected() const {
-    return !producer_edges.empty() || !consumer_edges.empty();
+    return !producer_edges.empty() || !consumer_edges.empty() || !output_vals.empty();
```

Comment:
@shmsong ^^^ this was dropping fusion group for me earlier. Stolen from Christian and seems to resolve the issue.

Reply:
LGTM. Thanks for fixing this.